### PR TITLE
Test for hero LWC

### PR DESCRIPTION
**Pull request by schandlergarcia**
### What does this PR do?

Add tests for the hero LWC

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### Functionality Before

hero was not being tested

### Functionality After

hero now has tests

### CODE REVIEW

File: force-app/main/default/lwc/hero/__tests__/hero.test.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('displays an image in the hero when the type is image', () => {
+        const element = createElement('c-hero', {
+            is: Hero
+        });
+        element.title = mockTitle;
```
Comment (pozil) [inline force-app/main/default/lwc/hero/__tests__/hero.test.js:31]: General FYI: no need to set all attributes if you're not testing them. Most of these are optional so you could greatly reduce the amount of code in your tests. Please do that so we can clearly see what's being tested.

For example, in this test you would only need to set `imgOrVideo` and `resourceUrl`.

File: force-app/main/default/lwc/hero/__tests__/hero.test.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const imageEl = element.shadowRoot.querySelector('img');
+            expect(imageEl).not.toBeNull();
+            // Verify that the URL returned matches, in the context of the test, the domain will render as http://localhost/
+            expect(imageEl.src).toBe(
+                'http://localhost/' + IMAGE_URL + mockResourceUrl
```
Comment (pozil) [inline force-app/main/default/lwc/hero/__tests__/hero.test.js:52]: Consider using string templates when concatenating strings:
```js
`http://localhost/${IMAGE_URL}${mockResourceUrl}`
```

File: force-app/main/default/lwc/hero/__tests__/hero.test.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        element.buttonClickProductOrFamilyName = mockButtonClickProductOrFamilyName;
+        document.body.appendChild(element);
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise rejects.
+        return Promise.resolve().then(() => {
+            // Select elements for validation
+            const divEl = element.shadowRoot.querySelector('div');
+            expect(divEl).not.toBeNull();
```
Comment (pozil) [inline force-app/main/default/lwc/hero/__tests__/hero.test.js:107]: also check for `overlay.style.opacity` value

File: force-app/main/default/lwc/hero/__tests__/hero.test.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            expect(heroDetailsEL.slogan).toBe(mockSlogan);
+            expect(heroDetailsEL.recordName).toBe(
+                mockButtonClickProductOrFamilyName
+            );
+            const spanEl = element.shadowRoot.querySelector('span');
+            expect(spanEl.textContent).toBe(mockButtonText);
+        });
+    });
+
+    it('displays the hero details component positioned right', () => {
```
Comment (pozil) [inline force-app/main/default/lwc/hero/__tests__/hero.test.js:149]: You're missing a test for default `heroDetailsPosition` value (centered hero).

File: force-app/main/default/lwc/hero/__tests__/hero.test.js (hunk 2 above)
Comment (schandlergarcia) [reply force-app/main/default/lwc/hero/__tests__/hero.test.js:52]: Good shout, I always forget about templates

File: force-app/main/default/lwc/hero/__tests__/hero.test.js (hunk 3 above)
Comment (schandlergarcia) [reply force-app/main/default/lwc/hero/__tests__/hero.test.js:107]: I'm looking for a good way to test the style, but not finding one. Do we have any examples?

Comment (pozil) [reply force-app/main/default/lwc/hero/__tests__/hero.test.js:107]: No particular example at hand but it should be pretty simple to test, you just need to extend that test:
1. Set opacity to `5` when creating the component
1. Add the following check at the end
```js
expect(divEl.style.opacity).toBe(0.5)
```

Comment (schandlergarcia) [reply force-app/main/default/lwc/hero/__tests__/hero.test.js:92]: Great, that's sorted!

**Comment by schandlergarcia** [comment]
@pozil this has now been updated!

**Pull request merged**